compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -407,6 +407,14 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             sym::transmute => {
                 self.copy_op_transmute(args[0], dest)?;
             }
+            sym::assert_inhabited => {
+                let ty = instance.substs.type_at(0);
+                let layout = self.layout_of(ty)?;
+
+                if layout.abi.is_uninhabited() {
+                    M::abort(self, format!("attempted to instantiate uninhabited type `{}`", ty))?;
```

Comment:
```suggestion
                    // The run-time intrinsic panics just to get a good backtrace; here we abort
                    // since there is no problem showing a backtrace even for aborts.
                    M::abort(self, format!("attempted to instantiate uninhabited type `{}`", ty))?;
```